src/pyhf/__init__.py

```diff
@@ -1,6 +1,6 @@
 from .tensor import BackendRetriever as tensor
 from .optimize import OptimizerRetriever as optimize
-from .version import __version__
+from .version import version as __version__
```

Comment:
```pytb
/home/runner/work/pyhf/pyhf/docs/cli.rst:4: WARNING: Failed to import "pyhf" from "pyhf.cli.cli". The following exception was raised:
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/sphinx_click/ext.py", line 390, in _load_module
    mod = __import__(module_name, globals(), locals(), [attr_name])
  File "/home/runner/work/pyhf/pyhf/src/pyhf/__init__.py", line 3, in <module>
    from .version import version as __version__
ModuleNotFoundError: No module named 'pyhf.version'
```

Reply:
> In docs, there are a few minor tweaks necessary to make sure the version is picked up correctly; just make sure you install the package and access it from there.

https://scikit-hep.org/developer/packaging

So I'll add an install bit to docs. Not sure how this will work for RTD though.

Reply:
I'm now more confused as we already install everything for the docs

https://github.com/scikit-hep/pyhf/blob/6b769fd6f5e1473deba2b4c55d49ebdb3db5b447/.github/workflows/docs.yml#L22-L32

Reply:
Why install setuptools and wheel? Those are only needed in the PEP 518 build environment, are they not?

Reply:
Also not sure why you have disabled the cache dir, this should be a clean build already? Ignore-installed, too?

Reply:
https://github.com/scikit-hep/pyhf/blob/dd946fc5ce895636a76eb299c6dc2f7f8bcf02cc/docs/conf.py#L22

If you want the src generated inplace, you can add `-e`, then `src/pyhf/version.py` will exist. But this looks really hacky/messy? Shouldn't you use the installed version not the hacked-in version in src? Isn't the point of src layout to _avoid_ being able to import without installing?

See for example @LovelyBuggies [config in Hist](https://github.com/scikit-hep/hist/blob/main/docs/conf.py), no path manipulations.

Reply:
> Why install setuptools and wheel? Those are only needed in the PEP 518 build environment, are they not?

Boilerplate laziness of me copy and pasting things between CI bits. Doesn't `wheel` work with `pip` though if a wheel isn't available from PyPI to build one locally for installs?

> Also not sure why you have disabled the cache dir, this should be a clean build already? Ignore-installed, too?

Why not? 

> But this looks really hacky/messy? Shouldn't you use the installed version not the hacked-in version in src? Isn't the point of src layout to avoid being able to import without installing?

Yeah, tracing back the `git blame` on this makes it seem like it was originally added near the start of the project in PR #157 and then we didn't properly clean this up and change it when we switched to a `src` directory. If I'm being honest the `docs/conf.py` needs a rework, but we've just left it as nothing has broke so far.

Reply:
> wheel isn't available from PyPI to build one locally for installs

Only if it's a "bad" package and doesn't have a pyproject.toml. Good packages with pyproject.toml's don't need them. Soonish, the backend for pip might be switched to use pep-517 by default, which will act like a pyproject.toml is there anyway. But yes, if you _depend on_ a wheel-less classic-build package, this is useful.

> Why not?

If you are building in CI, and there's no existing cache, it's unneeded to disable caching. And if you _add_ a cache to speed up your CI, then you'd have to go remove this too.

> but we've just left it as nothing has broke so far

Ugly code should die! Kill it if you have to! :)

Reply:
> But yes, if you _depend on_ a wheel-less classic-build package, this is useful.

Directly as a stated core dependency, no, but `pyrsistent` shows up in our CI given that it is a requirement of `jsonschema` which is a core dependency. https://pypi.org/project/pyrsistent/0.17.3/

> If you are building in CI, and there's no existing cache, it's unneeded to disable caching

Your suggestions are applied in PR #1453. I think when I first setup a lot of this I was still not actually believing that CI did what it said it would given experiences with Travis CI having versions of NumPy installed in the environments back in the day that you'd have to clear out before you could get your required versions in.

> Ugly code should die! Kill it if you have to! :)

No disagreement here. :) Just where it falls on the priority list. Feel free to send precocious students our way to make PRs to kill said code (I know you don't have extra time to be doing this for us).